sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -3028,6 +3032,15 @@ case class ConvertTimezone(
     sourceTs: Expression)
   extends TernaryExpression with ImplicitCastInputTypes with NullIntolerant {
 
+  def this(targetTz: Expression, sourceTs: Expression) = {
+    this(
+      Literal(SQLConf.get.sessionLocalTimeZone),
+      targetTz,
+      if (sourceTs.dataType == TimestampNTZType) sourceTs
+      else Cast(sourceTs, TimestampNTZType)
```

Comment:
How other types can appear if this below
```scala
override def inputTypes: Seq[AbstractDataType] = Seq(StringType, StringType, TimestampNTZType)
```
accepts only `TimestampNTZType`. Could add an end-to-end tests to `timestamp-ltz.sql` like
https://github.com/apache/spark/blob/master/sql/core/src/test/resources/sql-tests/inputs/timestamp-ntz.sql#L19